packages/redux/src/selectors.ts

```diff
@@ -41,38 +41,71 @@ export const getWorkspaceStorage = <State extends RootState>(
   store: State
 ): State['storage']['workspaceStorage'] => getStorage(store).workspaceStorage;
 
+export const getDefaultWorkspaceSettings = <State extends RootState>(
+  store: State
+): State['defaultWorkspaceSettings'] => store.defaultWorkspaceSettings;
+
 // Workspace
 export const getWorkspace = <State extends RootState>(
   store: State
-): State['workspace'] => store.workspace;
+): Workspace => {
+  const { workspace } = store;
+
+  if (workspace == null) {
+    // this will only be null on initial render
+    return null as never;
+  }
+  const defaultInjectedWorkspace: Workspace = {
+    data: {
+      ...workspace.data,
+      settings: getDefaultWorkspaceSettings(store) ?? {},
+    },
+  };
+  const customizedWorkspaceSettings = { ...workspace.data.settings };
+
+  const keys = Object.keys(
+    customizedWorkspaceSettings
+  ) as (keyof WorkspaceSettings)[];
+
+  for (let i = 0; i < keys.length; i += 1) {
+    const key = keys[i];
+    const value = customizedWorkspaceSettings[key];
+    // only set pass in customized defaults if defined
+    if (value !== undefined) {
+      defaultInjectedWorkspace.data.settings[key] =
+        value as WorkspaceSettings[typeof key] as never;
+    }
+  }
```

Comment:
This was because sometimes the userSettings have values intentionally set to undefined (not just missing) to let user reset values to default etc. setting a value to undefined means the value is still keyed in userSettings so doing object spreading would make those values undefined as opposed to the default.